app/subcommands/deploy_subcommand.py

```diff
@@ -267,21 +267,21 @@ def _non_registered_slaves(self, registered_slaves, slaves_to_validate):
         :return: list of slave hostnames that haven't registered with the master service yet
         :rtype: list[str]
         """
-        registered_rsa_keys = []
+        registered_host_ids = []
 
         for registered_slave in registered_slaves:
-            registered_rsa_keys.append(Network.rsa_key(registered_slave))
+            registered_host_ids.append(Network.get_host_id(registered_slave))
 
-        slaves_to_validate_rsa_key_host_pairs = {}
+        slaves_to_validate_host_id_pairs = {}
 
         for slave_to_validate in slaves_to_validate:
-            slaves_to_validate_rsa_key_host_pairs[Network.rsa_key(slave_to_validate)] = slave_to_validate
+            slaves_to_validate_host_id_pairs[Network.get_host_id(slave_to_validate)] = slave_to_validate
 
         non_registered_slave_hosts = []
 
-        for rsa_key in slaves_to_validate_rsa_key_host_pairs:
-            if rsa_key not in registered_rsa_keys:
-                non_registered_slave_hosts.append(slaves_to_validate_rsa_key_host_pairs[rsa_key])
+        for host_id in slaves_to_validate_host_id_pairs:
```

Comment:
I'm kind of beating a dead horse, but I think this sort of condition checking is more readable with a list comprehension. Maybe change the variables names so it can fit on one line.